src/controllers/api/accreditation.ts

```diff
@@ -0,0 +1,363 @@
+				return response.status(StatusCodes.BAD_REQUEST).send({
+					error: `${subjectDid} is deactivated`,
+				});
+			}
+
+			const resourceId = v4();
+			// construct credential request
+			const credentialRequest: CredentialRequest = {
+				subjectDid,
+				attributes: {
+					...attributes,
+					accreditedFor: schemas.map(({ url, type }: any) => ({
+						schemaId: url,
+						type,
+					})),
+					id: subjectDid,
+				},
+				issuerDid,
+				format: format || 'jwt',
+				connector: CredentialConnectors.Resource, // resource connector
+				credentialId: resourceId,
+				credentialName: accreditationName,
+			};
+
+			let resourceType: string;
+			switch (accreditationType) {
+				case AccreditationRequestType.authorize:
+					resourceType = DIDAccreditationTypes.VerifiableAuthorisationForTrustChain;
+					credentialRequest.type = [...(type || []), resourceType];
+					credentialRequest.termsOfUse = {
+						type: resourceType,
+						trustFramework: 'cheqd Governance Framework',
+						trustFrameworkId: 'https://learn.cheqd.io/governance/start',
+					};
+					break;
+				case AccreditationRequestType.accredit:
+					resourceType = DIDAccreditationTypes.VerifiableAccreditationToAccredit;
+					credentialRequest.type = [...(type || []), resourceType];
+					credentialRequest.termsOfUse = {
+						type: resourceType,
+						parentAccreditation,
+						rootAuthorization,
+					};
+					break;
+				case AccreditationRequestType.attest:
+					resourceType = DIDAccreditationTypes.VerifiableAccreditationToAttest;
+					credentialRequest.type = [...(type || []), resourceType];
+					credentialRequest.termsOfUse = {
+						type: resourceType,
+						parentAccreditation,
+						rootAuthorization,
+					};
+					break;
+			}
+
+			// validate parent and root accreditations
+			if (
+				accreditationType === AccreditationRequestType.accredit ||
+				accreditationType === AccreditationRequestType.attest
+			) {
+				const result = await AccreditationService.instance.verify_accreditation(
+					issuerDid,
+					parentAccreditation!,
+					true,
+					false,
+					response.locals.customer
+				);
+
+				if (result.success === false || result.data.rootAuthorization !== rootAuthorization) {
+					return response.status(StatusCodes.BAD_REQUEST).send({
+						error: `root Authorization or parent Accreditation is not valid`,
+					});
+				}
+			}
+
+			// issue accreditation
+			const accreditation: VerifiableCredential = await Credentials.instance.issue_credential(
+				credentialRequest,
+				response.locals.customer
+			);
+
+			// Track operation
+			const trackInfo = {
+				category: OperationCategoryNameEnum.CREDENTIAL,
+				name: OperationNameEnum.CREDENTIAL_ISSUE,
+				customer: response.locals.customer,
+				user: response.locals.user,
+				data: {
+					did: issuerDid,
+				} satisfies ICredentialTrack,
+			} as ITrackOperation;
+
+			eventTracker.emit('track', trackInfo);
+
+			return response.status(StatusCodes.OK).json({
+				didUrls: [
```

Comment:
Would be helpful to receive suggestions on the response schema

Currently it is
- didUrls [ didUrlwithPathparam, didUrlwithQueryParam ]
- accreditation

Reply:
For the query param version, is the name of the accreditation escaped correctly in case it has characters like spaces? Does it even allow spaces?

What happens if the credential issuance part is successful, but creating the resource fails?